modules/gapi/misc/python/test/test_gapi_infer.py

```diff
@@ -9,23 +9,205 @@
 
 class test_gapi_infer(NewOpenCVTests):
 
-    def test_getAvailableTargets(self):
-        targets = cv.dnn.getAvailableTargets(cv.dnn.DNN_BACKEND_OPENCV)
-        self.assertTrue(cv.dnn.DNN_TARGET_CPU in targets)
+    def infer_reference_network(self, model_path, weights_path, img):
+        net = cv.dnn.readNetFromModelOptimizer(model_path, weights_path)
+        net.setPreferableBackend(cv.dnn.DNN_BACKEND_INFERENCE_ENGINE)
+        net.setPreferableTarget(cv.dnn.DNN_TARGET_CPU)
+
+        blob = cv.dnn.blobFromImage(img)
+
+        net.setInput(blob)
+        return net.forward(net.getUnconnectedOutLayersNames())
+
+
+    def make_roi(self, img, roi):
+        return img[roi[1]:roi[1] + roi[3], roi[0]:roi[0] + roi[2], ...]
```

Comment:
How does this work?

Reply:
roi - is a tuple (x,y,h,w)

Reply:
> (x,y,h,w)

Probably it is `(x,y,w,h)`

Clear code (to avoid unclear error messages):

```
x, y, w, h = roi
return img[y:y+h, x:x+w, ...]
```